### PR TITLE
Continuation puzzle scene 1

### PR DESCRIPTION
**Pull request by AncientFungus**
Hopefully this means the first puzzle is completely done.

All textures and meshes work as they should.
All scripts work as they should.

There are extra stuff in this pull request that doesn't have anything to do with the first puzzle because everyone's had their hand in adding stuff to this branch.

I will make a separate branch for combining the scenes and that'll be our demo branch for monday.

Pull request made by Jens Lomander.

### CODE REVIEW

**Comment by andens** [comment]
iamheman

**Comment by andens** [comment]
The erroneous revert has itself been reverted. Note that some textures must be loaded again because this all happened during the migration to hct format.

**Comment by NinjaDanz3r** [comment]
iamheman

**Comment by NinjaDanz3r** [comment]
iamheman

File: src/Engine/Manager/ScriptManager.cpp [hunk 1]
```diff
@@ -696,9 +698,11 @@ void ScriptManager::Update(World& world, float deltaTime) {
 
                 if (script->IsInPropertyMap(name, typeId)) {
 
-                    if (typeId == engine->GetTypeIdByDecl("Entity@"))
+                    if (typeId == engine->GetTypeIdByDecl("Entity@")) {
+                        Entity* whatever = Hymn().GetEntityByGUID(*(unsigned int*)script->GetDataFromPropertyMap(name));
```
Comment (NinjaDanz3r) [inline src/Engine/Manager/ScriptManager.cpp]: MINOR Variable 'whatever' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 7 issues
* ![MINOR][MINOR] 7 minor

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L115](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/03c3f37ad07fabfe275b96dd850c069a09fd9388/src/Engine/Manager/ScriptManager.cpp#L115): Variable 'callstack' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L116](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/03c3f37ad07fabfe275b96dd850c069a09fd9388/src/Engine/Manager/ScriptManager.cpp#L116): Variable 'variables' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L604](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/03c3f37ad07fabfe275b96dd850c069a09fd9388/src/Engine/Manager/ScriptManager.cpp#L604): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L818](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/03c3f37ad07fabfe275b96dd850c069a09fd9388/src/Engine/Manager/ScriptManager.cpp#L818): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L824](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/03c3f37ad07fabfe275b96dd850c069a09fd9388/src/Engine/Manager/ScriptManager.cpp#L824): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Engine/Hymn.cpp [hunk 2]
```diff
@@ -225,12 +225,11 @@ void ActiveHymn::Render(Entity* camera, bool soundSources, bool particleEmitters
 }
 
 Entity* ActiveHymn::GetEntityByGUID(unsigned int GUID) {
-    const vector<Entity*>& entities = Hymn().world.GetEntities();
-    for (size_t i = 0; i < entities.size(); ++i) {
+    const std::vector<Entity*> entities = Hymn().world.GetEntities();
```
Comment (Chainsawkitten) [inline src/Engine/Hymn.cpp]: Should be a reference, like before.

File: src/Engine/Hymn.cpp.orig [hunk 3]
```diff
@@ -0,0 +1,247 @@
+#include "Hymn.hpp"
```
Comment (Chainsawkitten) [inline src/Engine/Hymn.cpp.orig]: Remove file

File: src/Engine/Manager/ScriptManager.cpp [hunk 4]
```diff
@@ -119,6 +119,7 @@ void AngelScriptDebugLineCallback(asIScriptContext* ctx, const std::map<std::str
 }
 
 void print(const std::string& message) {
+    std::cout << message;
```
Comment (Chainsawkitten) [inline src/Engine/Manager/ScriptManager.cpp]: We're already std::cout:ing Log messages.

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L115](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/9aaf8d11c5f8d94351541af8653ab5e8c1d2aa13/src/Engine/Manager/ScriptManager.cpp#L115): Variable 'callstack' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L116](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/9aaf8d11c5f8d94351541af8653ab5e8c1d2aa13/src/Engine/Manager/ScriptManager.cpp#L116): Variable 'variables' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L603](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/9aaf8d11c5f8d94351541af8653ab5e8c1d2aa13/src/Engine/Manager/ScriptManager.cpp#L603): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L815](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/9aaf8d11c5f8d94351541af8653ab5e8c1d2aa13/src/Engine/Manager/ScriptManager.cpp#L815): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L821](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/9aaf8d11c5f8d94351541af8653ab5e8c1d2aa13/src/Engine/Manager/ScriptManager.cpp#L821): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L115](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f9f131781a1970f127b4401c4219bbaddf69fb42/src/Engine/Manager/ScriptManager.cpp#L115): Variable 'callstack' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L116](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f9f131781a1970f127b4401c4219bbaddf69fb42/src/Engine/Manager/ScriptManager.cpp#L116): Variable 'variables' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L603](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f9f131781a1970f127b4401c4219bbaddf69fb42/src/Engine/Manager/ScriptManager.cpp#L603): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L815](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f9f131781a1970f127b4401c4219bbaddf69fb42/src/Engine/Manager/ScriptManager.cpp#L815): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L821](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f9f131781a1970f127b4401c4219bbaddf69fb42/src/Engine/Manager/ScriptManager.cpp#L821): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Chainsawkitten** [comment]
Since Mipmaps have been merged into master, you'll probably have to reimport all the textures when merging with master.

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L115](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/aaed47c6c4e7ea4243e82c23bf527968320cb2a8/src/Engine/Manager/ScriptManager.cpp#L115): Variable 'callstack' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L116](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/aaed47c6c4e7ea4243e82c23bf527968320cb2a8/src/Engine/Manager/ScriptManager.cpp#L116): Variable 'variables' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L603](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/aaed47c6c4e7ea4243e82c23bf527968320cb2a8/src/Engine/Manager/ScriptManager.cpp#L603): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L815](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/aaed47c6c4e7ea4243e82c23bf527968320cb2a8/src/Engine/Manager/ScriptManager.cpp#L815): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L821](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/aaed47c6c4e7ea4243e82c23bf527968320cb2a8/src/Engine/Manager/ScriptManager.cpp#L821): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**